internal/commands/push/errors.go

```diff
@@ -1,10 +1,19 @@
 package push
 
-type errProjectNotFound struct {
+import (
+	"fmt"
+)
+
+type errProjectInvalid struct {
+	Path   string // app project path
+	Exists bool   // indicates if the path exists
 }
 
-func (err errProjectNotFound) Error() string {
-	return "must specify --local or run command from inside a Realm app directory"
+func (err errProjectInvalid) Error() string {
+	if err.Exists { // if path exists but is not a realm app directory
```

Comment:
you could probably omit the comment and make this the `if !err.Exists` condition, and everything would still read fairly clearly (but carry on as you see fit here)